### ISSUE TITLE
`jest/valid-mock-module-path` does not work with typescript path aliases

### ISSUE THREAD

**Issue by atheck** [jest-community/eslint-plugin-jest#1857]
Having path aliases in the tsconfig.json file, like:

```jsonc
// ...
  "paths": {
      "@lib/*": [
        "./path/to/lib/*"
      ],
  }
// ...

the "valid-mock-module-path" rule flags a call to `jest.mock("@lib/someModule");`.

I think this is a false positive.

**Comment by G-Rath** [comment]
yup we totally should support a `aliases` kind of option

**Comment by hainenber** [comment]
Appreciate the rule usage @atheck! 

@G-Rath  implementation wise, I wonder what would be the most proper approach for this issue? Should we have a rule option like `aliases`, duplicating the config? Or we should rely on `typescript` installation to do the needful, making the rule bit more cumbersome by looping in TS machineries?

**Comment by atheck** [comment]
I think it should be an option because bundlers like babel and webpack also support aliases. So you can use aliases without typescript and eslint can't know where to look for the correct aliases configuration. I think copying the configuration would be fine here.

**Comment by G-Rath** [comment]
yeah what @atheck said, especially since going forward ESLint configs are always JS it should be trivial for people to pass through their options e.g. `require('./tsconfig.json')`.

The option we're actually mirroring here is the [`moduleNameMapper`](https://jestjs.io/docs/configuration#modulenamemapper-objectstring-string--arraystring), though we can just start with a simple `Record<string, string>` (it might be useful to support regexps, but again people can do logic in their config files, and it's a breaking change for us to change that feature afterwards, so lets stick with just strings for now).

I would expect if we ever supporting reading Jest config files, we'd use that property by default

tl;dr start with a `Record<string, string>` which we can just "`if (p in aliases)`" as part of the rule

**Comment by jerone** [comment]
Automatic detection for aliases would have my preference. The less configuration on double places the better.

We are using another ESLint plugin that detects `tsconfig.json` and `jsconfig.json` aliases automatically. In this case for imports. Maybe you can find inspiration from this package: https://github.com/Limegrass/eslint-plugin-import-alias
